test/explorer/data_frame_test.exs

```diff
@@ -8,6 +8,19 @@ defmodule Explorer.DataFrameTest do
     {:ok, df: Explorer.Datasets.fossil_fuels()}
   end
 
+  setup do
+    :ok = File.touch!("tmp_test.csv")
+
+    on_exit(fn ->
+      File.rm!("tmp_test.csv")
+    end)
+  end
+
+  def tmp_csv(contents) do
+    :ok = File.write!("tmp_test.csv", contents)
+    "tmp_test.csv"
+  end
```

Comment:
It is better to write to a tmp directory. Since this requires Elixir v1.12, we can leverage the `tmp_dir` tag:

```elixir
@tag :tmp_dir
test "my test", config do
  csv = tmp_csv(config, """
  ...
  """)
end
```

where `tmp_csv` is:

```elixir
defp tmp_csv(config, contents) do
  path = Path.join(config.tmp_dir, "tmp.csv")
  :ok = File.write!(path, contents)
  path
end
```

Reply:
TIL! I must have missed the `tmp_dir` tag in all the other stuff to come out with 1.12.